raw_map/src/transform/merge_short_road.rs

```diff
@@ -125,6 +126,13 @@ impl RawMap {
                 assert_eq!(r.i2, i2);
                 new_id.i2 = i1;
             }
+
+            if new_id.i1 == new_id.i2 {
```

Comment:
Example is https://www.openstreetmap.org/node/330683931. We merge all the short segments here and produce a nice single intersection. But previously, one of the intermediate states was a tiny loop road that was getting filtered out when `InitialMap` was made. Simpler to immediately filter it out here.